src/services/procosysApi.ts

```diff
@@ -37,41 +37,145 @@ import {
     SavedSearch,
     PunchItemSavedSearchResult,
     ChecklistSavedSearchResult,
+    isPlants,
 } from './apiTypes';
 
 type ProcosysApiServiceProps = {
-    axios: AxiosInstance;
+    baseURL: string;
     apiVersion: string;
+    cb2?: (res: Response) => Response;
 };
 
-const typeGuardErrorMessage = (expectedType: string): string => {
+type GetOperationProps = {
+    abortSignal?: AbortSignal;
+    method: string;
+    headers: any;
+    responseType?: string;
+};
+
+export const typeGuardErrorMessage = (expectedType: string): string => {
     return `Unable to retrieve ${expectedType}. Please try again.`;
 };
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
-const procosysApiService = ({ axios, apiVersion }: ProcosysApiServiceProps) => {
+const procosysApiService = (
+    {
+        baseURL,
+        apiVersion,
+        cb2 = (res: Response): Response => res,
+    }: ProcosysApiServiceProps,
+    token: string
+) => {
     // General
     const getVersion = (): string => {
         return apiVersion;
     };
+
+    const getByFetch = async (
+        url: string,
+        abortSignal?: AbortSignal,
+        additionalHeaders?: any,
+        isBlob?: boolean
+    ): Promise<any> => {
+        const GetOperation: GetOperationProps = {
+            method: 'GET',
+            headers: {
+                Authorization: `Bearer ${token}`,
+                ...additionalHeaders,
+            },
```

Comment:
Have you tried adding abortSignal to the getOperation directly, without checking whether it's unefined or not? If the http call works while sendning undefined as abortSignal, then there is no need to check whether the abort signal exists.